src/imitation/rewards/reward_net.py

```diff
@@ -1,30 +1,29 @@
 """Constructs deep network reward models."""
 
 from abc import ABC, abstractmethod
-from typing import Iterable, Optional, Sequence, Tuple
+from typing import Optional
 
 import gym
-import tensorflow as tf
+import numpy as np
+import torch as th
+from stable_baselines3.common.preprocessing import get_flattened_obs_dim
+from torch import nn
 
-from imitation.util import networks, serialize
+import imitation.rewards.common as rewards_common
+from imitation.util import networks
 
 
-class RewardNet(serialize.Serializable, ABC):
+class RewardNet(nn.Module, ABC):
     """Abstract reward network.
 
-    This class assumes that the caller will set the default TensorFlow Session
-    and initialize the network's variables.
-
     Attributes:
-      observation_space: The observation space of `obs_ph` and `next_obs_ph`.
-      action_space: The action space of `act_ph`.
-      obs_ph (tf.Tensor): previous observation placeholder.
-      act_ph (tf.Tensor): action placeholder.
-      next_obs_ph (tf.Tensor): next observation placeholder.
-      _params (dict): parameters to serialize in `save`, used as keyword
-          arguments for constructor by `load`.
-      _layers (dict): CheckpointableBase objects, e.g. a TensorFlow layer,
-          saved by `save` and restored by `load`.
+      observation_space: The observation space.
+      action_space: The action space.
+      use_state: should `base_reward_net` pay attention to current state?
```

Comment:
Any particular reason for making these attributes of the base class? I'm fine with this design choice, just curious if there's a particular use case for it.

Reply:
I assumed that they would be necessary for all subclasses, but that different subclasses would have different ways of implementing them (e.g. if you have image-based observations and categorical actions, then you might want to feed the action into an MLP after the conv layers). In any case, it's going to get deleted in a future PR.